apps/api/src/service/collaboratorsService.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { type PostgresDb } from '@/db/index.js';
+import { collaborators } from '@/db/schemas/collaborators.js';
+import logger from '@/logger.js';
+import { type AsyncResult, failure, success } from '@/utils/results.js';
+import { and, eq } from 'drizzle-orm';
+import { type CollaboratorModel, type CollaboratorRecord } from './types.js';
+
+const collaboratorsSvc = (db: PostgresDb) => ({
+	createCollaborators: async ({
+		newCollaborators,
+	}: {
+		newCollaborators: CollaboratorModel[];
+	}): AsyncResult<CollaboratorRecord[]> => {
```

Comment:
Should this accept transaction as parameter?

Reply:
Potentially, there's currently no use case for creating Collaborators simultaneously with another DB action